### PR TITLE
Code sample tests

### PR DESCRIPTION
**Pull request by Simply007**
### Motivation

Part of #96

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

**Comment by Simply007** [comment]
I would like to discuss specifically https://github.com/Kentico/kontent-management-sdk-net/pull/101/commits/ef828f352987f5f8755c3e9cbf3f446f52607282 @gormal.

It eases the work with dynamic element in language variants and components, but require using `ToDynamic()` which is easy to forget (because there is no buildtime check). There might be some way to overload an implicit conversion to `dynamic` that I am not aware of to at least fire some warning.

Example could be seen in `Kentico.Kontent.Management.Tests.Unit.CodeSamples.CmApiV2.PutLanguageVariant`

**Comment by Simply007** [comment]
> I would like to discuss specifically [ef828f3](https://github.com/Kentico/kontent-management-sdk-net/commit/ef828f352987f5f8755c3e9cbf3f446f52607282) @gormal.
> 
> It eases the work with dynamic element in language variants and components, but require using `ToDynamic()` which is easy to forget (because there is no buildtime check). There might be some way to overload an implicit conversion to `dynamic` that I am not aware of to at least fire some warning.
> 
> Example could be seen in `Kentico.Kontent.Management.Tests.Unit.CodeSamples.CmApiV2.PutLanguageVariant`

I have tested out rewriting the Elements property to be IEnumerable<BaseElement> without any success - see #104

**Comment by Simply007** [comment]
It should be ready - let's discuss @gormal whet alternative of upserting language variant we want to have in code samples.

https://github.com/KenticoDocs/kontent-docs-samples/pull/39


I vote for element builder ones.

**Comment by Simply007** [comment]
* [ ] - discuss whether `Kentico.Kontent.Management.Models.LanguageVariants.Elements.BaseElement.GetDynamicReference` Should call ToString for `Reference.Id`.

File: Kentico.Kontent.Management.Tests/Unit/CodeSamples/CmApiV2.cs [hunk 1]
```diff
@@ -0,0 +1,1385 @@
...
+
+        // DocSection: cm_api_v2_delete_variant
+        // Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
+        [Fact]
+        public async void DeleteLanguageVariant()
+        {
+            var client = _fileSystemFixture.CreateMockClient(Substitute.For<IManagementHttpClient>());
+
+            var identifier = new LanguageVariantIdentifier(Reference.ById(Guid.Parse("f4b3fc05-e988-4dae-9ac1-a94aba566474")), Reference.ById(Guid.Parse("d1f95fde-af02-b3b5-bd9e-f232311ccab8")));
+            // var identifier = new LanguageVariantIdentifier(Reference.ById(Guid.Parse("f4b3fc05-e988-4dae-9ac1-a94aba566474")), Reference.ByCodename("es-ES"));
```
[the remark this answers is not in the document]
Comment (Simply007) [reply Kentico.Kontent.Management.Tests/Unit/CodeSamples/CmApiV2.cs:135]: yeap - I use these parts in code samples to showcase identifier construction possibilities. It is the same for JS and Curl as well

File: Kentico.Kontent.Management/Modules/ModelBuilders/ElementBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    /// Provides extension methods for building dynamic elements from strongly typed element based on the <see cref="BaseElement"/>.
+    /// </summary>
+    public static class ElementBuilder
+    {
+        /// <summary>
+        /// Builds a dynamic elements enumeration from a strongly typed elements based on the <see cref="BaseElement"/>.
+        /// </summary>
+        public static IEnumerable<dynamic> GetElementsAsDynamic(params BaseElement[] elements)
+        {
+            elements.Where(elementObject => elementObject.Element == null).ToList().ForEach(element =>
```
[the remark this answers is not in the document]
Comment (Simply007) [reply Kentico.Kontent.Management/Modules/ModelBuilders/ElementBuilder.cs:19]: Yeap - I might need some guidance how to make it pretty and in what format to return an exception with all of the data.

[the remark this answers is not in the document]
Comment (Simply007) [reply Kentico.Kontent.Management/Modules/ModelBuilders/ElementBuilder.cs]: 1d967a2b

**Review by gormal — APPROVED** [review]

**Pull request merged**